dev_guide/builds.adoc

```diff
@@ -1557,3 +1557,84 @@ Docker and Source builds set the following labels on output images:
 |*io.openshift.build.source-location*
 |Source URL for the build
 |===
+
+
+[[using-external-artifacts]]
+= Using External Artifacts During The Build
+
+In order to use external artifacts during the build you don't need any special
+support from OpenShift platform. You could download and use any artifacts
+as you want.
+
+For `*Source*` build strategy you need to put appropriate shell commands into
+the `assemble` script.
+
+.`.sti/bin/assemble` file
+====
+
+[source,bash]
+----
+#!/bin/sh -x
+wget http://code.jquery.com/jquery-2.1.4.min.js
+----
+====
+
+For `*Docker*` build strategy you need to modify `Dockerfile` and invoke shell
+commands with https://docs.docker.com/engine/reference/builder/#run[`RUN` instruction].
+
+.Excerpt of `Dockerfile`
+====
+
+[source]
+----
+RUN wget http://code.jquery.com/jquery-2.1.4.min.js
+----
+====
+
+Both of the examples will work, but what if you don't want to have a hard coded
+version of the artifact?
+
+In this case you should extract version from the URL and replace it by the
+environment variable (in our example it could be named `JQUERY_VERSION`).
+
+Now let’s recall and summarize how to manage these environment variables.
+If you are using `*Source*` strategy you could choose between the following ways:
+
+.  link:#environment-files[put them into the `.sti/environment` file]
++
+[source,shell]
+----
+JQUERY_VERSION=2.1.4
+----
+
+.  link:#buildconfig-environment[set them in `*BuildConfig*`]
++
+[source,yaml]
+----
+  strategy:
+    sourceStrategy:
+      env:
+      - name: JQUERY_VERSION
+        value: "2.1.4"
+----
+
+.  provide them explicitly with `*oc start-build --env*` (only for builds that are triggered manually)
++
+[source,shell]
+----
+oc start-build --env=JQUERY_VERSION=2.1.4 <BuildConfigName>
+----
+
+Also, consider one more way: you could link:#override-builder-image-scripts[create your custom version of the `assemble` script]
+and use it instead of the default or from the repository.
```

Comment:
Hmm, but reading 

> Also, consider one more way: you could create your custom version of the `assemble` script and use it instead of the default or from the repository.

That doesn't suggest what you're explaining here in the comments.